### DISCUSSION TITLE
"+1" comment warning

### DISCUSSION THREAD

**Discussion by charles-allen** [community/community#4921] [Projects and Issues] [answered]
## Problem

- It's tedious reading through lists of comments that just say "+1"
- GitHub has emotes, and this appears to be the "agreed" way to upvote an issue
- IMO the issue persists because people don't understand this convention

## Solution

When a user attempts to post "+1" (or simple derivatives):
- Show a message (perhaps with a link to docs):  
  > GitHub convention is to upvote issues or comments using emotes.  
  > Please avoid adding comments that only exist to upvote.
- Optional: the dialog *could* have 2 buttons: "Ok" (default) and "Post anyway" (red), to override false-positives

**Comment by lukehefson** [comment]
Hey @charles-allen! They aren't quite as targeted as the solutions you’ve proposed above– but you might like to know that there are a couple of things that GitHub does to attempt to dissuade folks from doing needless `+1`s on the platform:

| If we detect a solitary +1 (or similar) comment we show a tooltip | <img width="342" alt="Screenshot 2021-08-12 at 16 34 08" src="https://user-images.githubusercontent.com/1469659/129551954-6f7ca030-708f-46c7-a0fc-a60e9fa880d9.png"> |
|--------|--------|
| **If a thread has more than 3 comments with exactly the same content – we truncate those**  | <img width="611" alt="Screenshot 2021-08-12 at 16 34 42" src="https://user-images.githubusercontent.com/1469659/129552177-87d14a3c-1810-471d-82c8-be58f9358ef4.png"> |

**Reply by charles-allen** [reply]
I didn't know about these, but given that these features were already active prior to my post, I would say that they aren't effective enough (at least for my taste). That said, I am definitely on the fussy end of the scale!

Another potential avenue that I didn't think of previously might be to fix it on the viewer's side, ex: allow users to elect for more aggressive truncation.

I've lost track of the specific thread that prompted my post -- next time I come across a good example I will share it here :) Specifics should help narrow down a good solution!

**Reply by hughlilly** [reply]
> **If a thread has more than 3 comments with exactly the same content – we truncate those**	<img alt="Screenshot 2021-08-12 at 16 34 42" width="611" src="https://user-images.githubusercontent.com/1469659/129552177-87d14a3c-1810-471d-82c8-be58f9358ef4.png">

@lukehefson, if the code is detecting “the exact same content,” the UI writing should reflect this: "3 identical comments".

However, if there is something more going on and the comments are genuinely merely _similar_, the current wording is fine—but I would question the benefit of hiding messages that are not identical.

**Reply by michaelblyons** [reply]
In addition to "+1," honorable mention to `(I|we) need this[[:punct:]]?(to+[[:punct:]]*)?` and `(devs,?\s+)?(please|plz)?\s*add this`.

**Comment by cynicaljoy** [comment, marked as answer]
What about automatically converting "+1" comments to a 👍 reaction on the initial post and displaying an info prompt letting the user know that their message has automatically been converted? I know it's a little magical and users could be confused, but I think it's worth the risk for the rest of our sanity 😁

**Reply by charles-allen** [reply]
I almost suggested this, but I thought it might be too invasive. But actually, instead of telling the user how to do it, showing them is surely better! The post could disappear, the page scroll up, and the new 👍 could be flashing.

I guess the explanation prompt could allow some sort of "cancel"; perhaps it could be labeled "continue editing".

**Reply by cynicaljoy** [reply]
I was thinking the info box would show up up the top of the page letting them know their comment was automatically converted to a reaction. I think this is completely acceptable in cases when their message is _only_ "+1"

**Reply by Davidharvey123** [reply]
https://github.com/Davidharvey123/github-slideshow/issues/1

**Reply by hughlilly** [reply]
I came here to suggest exactly this. Automatically convert and display a message—even a dismissable banner—to that effect.

**Reply by hughlilly** [reply]
@charles-allen:
> I guess the explanation prompt could allow some sort of "cancel"; perhaps it could be labeled "continue editing”.

I don’t think there’s any benefit in allowing the user to continue editing. Removing a reaction is as simple as clicking on it, which is intuitive enough.

**Reply by charles-allen** [reply]
> @charles-allen:
> > I guess the explanation prompt could allow some sort of "cancel"; perhaps it could be labeled "continue editing”.
> 
> I don’t think there’s any benefit in allowing the user to continue editing. Removing a reaction is as simple as clicking on it, which is intuitive enough.
> 
> 

I was more thinking about retrieving the converted message. It depends how strict the "+1 detection" logic is. For example it might be difficult to distinguish between waffly +1s and +1s that also include an important comment:
- "+1, nice job _Dave_!"
- "+1, don't forget _XYZ_"

**Reply by hughlilly** [reply]
Ah, I was thinking of posts that contain _only_ “+1" and nothing else. Extended +1s, for want of a term, ought to be allowed through, in my opinion.

**Reply by charles-allen** [reply]
> Ah, I was thinking of posts that contain _only_ `+1` and nothing else.

You're right; it's better to just focus on a v1 of this feature that only deals with "+1". That's probably >90% of the +1 spam, and then the UX is simple => just convert them to 👍

**Comment by koesper** [comment]
The "+1" spam is still a problem, the current solutions don't seem very effective.
Also, the 'we'll collapse the thread' function doesnt stop notifications from being sent, i still get lots of emails notifying me of someone just posting a silly "+1" (or the variations like "+99" messages).

Perhaps the gentle tooltip should be replaced by a confirmation dialog, explaining why using a 👍 reaction is better, and less annoying?

[This](https://github.com/orgs/community/discussions/56979) is a good example of a thread where the actual content gets snowed under by lots of "+1" spam

**Comment by howsiyu** [comment]
[This](https://github.com/microsoft/vscode/issues/196740) is another example with a lot of comments just spamming "same issue here".

**Comment by traviswimer** [comment]
Something that could also help with any form of "comment spam" is to make the 👎 reaction actually notify the commenter. Give them a little taste of their own medicine.
